acl2_kernel/kernel.py

```diff
@@ -38,7 +38,18 @@ def __init__(self, **kwargs):
     def _start_acl2(self):
         sig = signal.signal(signal.SIGINT, signal.SIG_DFL)
         try:
-            self.acl2wrapper = replwrap.REPLWrapper(os.environ['ACL2'], 'ACL2 !>', None)
+            prompt_change_cmd = '''
+(progn
+  (set-state-ok t)
+  (defun jupyter-prompt (channel state)
+    (declare (xargs :mode :program))
+    (fmt1 "JPY-ACL2>" '() 0 channel state nil))
+  (set-state-ok nil))
+(set-ld-prompt 'jupyter-prompt state)
+(reset-prehistory)'''
```

Comment:
This is just for my interests. What line prevents changing the prompt after changing the prompt to `JPY-ACL2>`

Reply:
The line

```lisp
(set-ld-prompt 'jupyter-prompt state)
```

prevents changing of the prompt by installing a prompt that is independent of the ACL2 environment.

Of course, an acl2-kernel user can change the prompt inside a notebook by using `set-ld-prompt` but I don't see any reason for doing so. Besides that, even if I were to redefine `set-ld-prompt`, one can have a look at the body of `set-ld-prompt` using `:pe set-ld-prompt` and then follow what is stated.

Reply:
I see. This is nice explanation. Thanks